src/psm/src/pdnsim.cpp

```diff
@@ -116,7 +117,9 @@ void PDNSim::analyzePowerGrid(odb::dbNet* net,
 
   last_corner_ = corner;
   auto* solver = getIRSolver(net, false);
-  solver->solve(corner, source_type, voltage_source_file);
+  if (!use_prev_solution || !solver->hasSolution(corner)) {
```

Comment:
I think it would be good to have an else with an info msg so that people know reuse actually happened.